pkg/templates/contents.go

```diff
@@ -20,125 +20,129 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/http"
+	"log"
 	"os"
+	"path"
 	"path/filepath"
-	"strings"
 
-	"github.com/google/go-github/v41/github"
-	"golang.org/x/oauth2"
+	"github.com/hashicorp/go-getter"
+	"gopkg.in/yaml.v2"
+
+	"github.com/nitrictech/newcli/pkg/utils"
 )
 
-type RepoContent interface {
-	ListSubDirectories(path string) ([]string, error)
-	DownloadDirectoryContents(path string, destDirectory string, force bool) error
+type Template struct {
+	Name string `yaml:"name"`
+	Path string `yaml:"path"`
 }
 
-type repoContent struct {
-	client *github.Client
-	repo   string
-	owner  string
+type TemplatesConfig struct {
+	Templates []Template
 }
 
-func NewRepoContent(owner, repo string) RepoContent {
-	ctx := context.Background()
+const RAW_GITHUB_URL = "https://raw.githubusercontent.com"
 
-	token := os.Getenv("GITHUB_AUTH_TOKEN")
-	var tc *http.Client
-	if token != "" {
-		ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
-		tc = oauth2.NewClient(ctx, ts)
-	} else {
-		fmt.Println("Attempting to download from github without a GITHUB_AUTH_TOKEN")
-	}
+const TEMPLATES_REPO = "nitrictech/stack-templates"
 
-	return &repoContent{
-		repo:   repo,
-		owner:  owner,
-		client: github.NewClient(tc),
-	}
-}
+const TEMPLATES_REPO_GIT_URL = "github.com/nitrictech/stack-templates.git"
+
+var CONFIG_PATH = path.Join(utils.NitricHome(), "store", "repositories.yml")
 
-func (r *repoContent) ListSubDirectories(path string) ([]string, error) {
-	_, directoryContent, _, err := r.client.Repositories.GetContents(context.Background(), r.owner, r.repo, path, nil)
+func ReadTemplatesConfig() TemplatesConfig {
+	var config TemplatesConfig
+
+	// Open YAML file
+	file, err := os.Open(CONFIG_PATH)
 	if err != nil {
-		return nil, err
+		log.Println(err.Error())
 	}
+	defer file.Close()
 
-	var subdirs []string
-	for _, c := range directoryContent {
-		switch *c.Type {
-		case "file":
-		case "dir":
-			if !strings.HasPrefix(*c.Path, ".") && strings.Contains(*c.Path, "-stack") {
-				subdirs = append(subdirs, *c.Path)
-			}
+	// Decode YAML file to struct
+	if file != nil {
+		decoder := yaml.NewDecoder(file)
+		if err := decoder.Decode(&config); err != nil {
+			log.Println(err.Error())
 		}
 	}
-	return subdirs, nil
+
+	return config
 }
 
-func (r *repoContent) DownloadDirectoryContents(path string, destDir string, force bool) error {
-	_, err := os.Stat(destDir)
-	if err == nil && !force {
-		return errors.New("stack directory already exists and isn't empty, choose a different name or use the --force flag to create in a non-empty directory")
-	}
+func GetOfficialStackName(name string) string {
```

Comment:
Does this need to be public? if a symbol starts with a capital it is public. so consider making this.
Also avoid starting fuctions with "get", see https://go.dev/doc/effective_go#Getters
officialStackName()